### PR TITLE
MTKA-1265: Make relationship fields with back references editable from the reverse side

### PR DESCRIPTION
**Pull request by matthewguywright**
## Description
Allows the display of reverse relationships on the publisher entry screen.

### CODE REVIEW

**Review by ChrisWiegman — DISMISSED** [review]

**Comment by nickcernis** [comment]
@mindctrl @matthewguywright I completed tests above in 8abf508ec1b8035c527c401e33f1f75f81a37c71. These exclude tests for saving data, which needs to be fixed as part of MTKA-1291.

This is ready to review.

I suggest we do not merge to main until MTKA-1291 is also ready to merge (fixing reverse relationship fields so they save data). That way we keep main in a deployable state. I'm working on 1291 now.

File: tests/acceptance/CreateRelationshipFieldEntryCest.php [hunk 1]
```diff
@@ -3,35 +3,14 @@
...
-		$i->wait( 1 );
-		$i->fillField( [ 'name' => 'name' ], 'Many Companies' );
-		$i->selectOption( '#reference', 'Companies' );
-		$i->click( 'input#many-to-many' );
-		$i->click( '.open-field button.primary' );
-		$i->wait( 1 );
 	}
 
+	/**
+	 * @before create_company_employee_models
```
Comment (mindctrl) [inline tests/acceptance/CreateRelationshipFieldEntryCest.php:12]: Never noticed this feature before... nice 😎

**Review by mindctrl — DISMISSED** [review]

**Comment by nickcernis** [comment]
The PHP unit test fail after the recent merge here seems to relate to WP requiring a new third-party dependency to run unit tests under PHPUnit 8.

Error:

```
> phpunit
Error: The PHPUnit Polyfills library is a requirement for running the WP test suite.
If you are trying to run plugin/theme integration tests, make sure the PHPUnit Polyfills library (https://github.com/Yoast/PHPUnit-Polyfills) is available and either load the autoload file of this library in your own test bootstrap before calling the WP Core test bootstrap file; or set the absolute path to the PHPUnit Polyfills library in a "WP_TESTS_PHPUNIT_POLYFILLS_PATH" constant to allow the WP Core bootstrap to load the Polyfills.
```

References:
- https://make.wordpress.org/core/2021/09/27/changes-to-the-wordpress-core-php-test-suite/ (includes a section titled “Step-by-step: how to make your test setup compatible with these changes and with higher PHPUnit versions”).
- https://core.trac.wordpress.org/ticket/53945
- https://github.com/Yoast/PHPUnit-Polyfills

This may also affect other branches when tests run there.

**Comment by mindctrl** [comment]
This PR relies on #353 and should not be merged until that one is merged into here.

**Comment by nickcernis** [comment]
This is now ready for a final review.

**Review by nickcernis — APPROVED** [review]

**Review by mindctrl — APPROVED** [review]

**Pull request merged**